base/db/R/query.traits.R

```diff
@@ -21,26 +21,17 @@
 ##' @export query.traits
 ##' @examples
 ##' \dontrun{
+##' con <- db.open(your_settings_here)
 ##' species <- query.pft_species('ebifarm.c4crop')
 ##' spstr <- vecpaste(species$id)
 ##' trvec <- c('leafN', 'SLA')
-##' trait.data <- query.traits(spstr, trvec)
+##' trait.data <- query.traits(spstr, trvec, con)
 ##' }
 ##' @author David LeBauer, Carl Davidson, Shawn Serbin
-query.traits <- function(ids, priors, con = NULL,
-                         update.check.only=FALSE,
-                         ids_are_cultivars=FALSE){
-  
-  if(is.null(con)){
-    con <- db.open(settings$database$bety)
-    on.exit(db.close(con), add = TRUE)
-  }
-  if(is.list(con)){
-    print("query.traits")
-    print("WEB QUERY OF DATABASE NOT IMPLEMENTED")
```

Comment:
@infotroph I don't recall exactly what set of conditions was resulting in a different type of connection object being passed to this function, but I recall it being a pain to track down (cryptic message) which is why we created this message. This was different from the case you removed, of no connection being passed in (i.e. open connection was passed it, but it wasn't in a form we could use)

Reply:
@mdietze I'll add it back. Three questions:
* Keep testing for list specifically, or convert it to a check for anything that's not an (open, usable) connection?
* Wording suggestions for the error message, or for a comment in the code to help future editors?
* logger.error or logger.severe?

Reply:
@mdietze does d49ad92ba fix this?